plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHivePartitionedBucketFunction.java

```diff
@@ -0,0 +1,133 @@
+            bucketPositions.computeIfAbsent(hiveBucket, ignored -> new ArrayList<>()).add(i);
+        }
+
+        BucketFunction hivePartitionedBucketFunction = partitionedBucketFunction(hiveBucketingVersion, 10, ImmutableList.of(HIVE_LONG), ImmutableList.of(BIGINT), 100);
+        // All positions of a bucket should hash to the same partitioned bucket
+        for (Map.Entry<Integer, List<Integer>> entry : bucketPositions.entrySet()) {
+            assertEquals(
+                    entry.getValue().stream()
+                            .map(position -> hivePartitionedBucketFunction.getBucket(page, position))
+                            .distinct()
+                            .count(),
+                    1);
+        }
+    }
+
+    @Test(dataProvider = "hiveBucketingVersion")
+    public void testMultiplePartitions(BucketingVersion hiveBucketingVersion)
+    {
+        int numValues = 1024;
+        Block bucketColumn = createLongSequenceBlock(923402935, 923402935 + numValues);
+        Page bucketedColumnPage = new Page(bucketColumn);
+
+        int numPartitions = 8;
+        ImmutableList.Builder<Long> partitionValuesBuilder = ImmutableList.builder();
+        for (int i = 0; i < numPartitions; i++) {
+            partitionValuesBuilder.addAll(Collections.nCopies(numValues / numPartitions, i * 348349L));
+        }
+        List<Long> partitionValues = partitionValuesBuilder.build();
+        Block partitionColumn = createLongsBlock(partitionValues);
+        Page page = new Page(bucketColumn, partitionColumn);
+        BucketFunction hiveBucketFunction = bucketFunction(hiveBucketingVersion, 10, ImmutableList.of(HIVE_LONG));
+        Map<SimpleImmutableEntry<Integer, Long>, List<Integer>> partitionedBucketPositions = new HashMap<>();
+
+        for (int i = 0; i < numValues; i++) {
+            int hiveBucket = hiveBucketFunction.getBucket(bucketedColumnPage, i);
+            long hivePartition = partitionValues.get(i);
+            // record list of positions for each combination of hive partition and bucket
+            partitionedBucketPositions.computeIfAbsent(
+                    new SimpleImmutableEntry<>(hiveBucket, hivePartition),
+                    ignored -> new ArrayList<>()).add(i);
+        }
+
+        BucketFunction hivePartitionedBucketFunction = partitionedBucketFunction(hiveBucketingVersion, 10, ImmutableList.of(HIVE_LONG), ImmutableList.of(BIGINT), 100);
+        // All positions of a hive partition and bucket should hash to the same partitioned bucket
+        for (Map.Entry<SimpleImmutableEntry<Integer, Long>, List<Integer>> entry : partitionedBucketPositions.entrySet()) {
+            assertEquals(
+                    entry.getValue().stream()
+                            .map(position -> hivePartitionedBucketFunction.getBucket(page, position))
+                            .distinct()
+                            .count(),
+                    1);
+        }
```

Comment:
Please also check that `hivePartitionedBucketFunction` generates `8 * 10` distinct values

Reply:
I've added this test, but I found that I had to tweak bucket count of the partitionedBucketFunction to 4000 to get 80 distinct buckets. It seems even though we generate 80 unique hash values the bucket count has to be some larger value, otherwise the `% N` operation results in 60-70 distinct values.

Reply:
I don't see 4000 anywhere. Just increase the number of rows in block?
I we cannot fill 80 buckets with `1024` it seems that the bucket function is not that great.

Reply:
It's on line 111 of this class.
Increasing no. of values doesn't make a difference if the partition and bucket values are still the same